generator/generater.go

```diff
@@ -11,26 +11,36 @@ import (
 	"github.com/Yamashou/gqlgenc/config"
 )
 
-// mutateHook adds the "omitempty" option to nilable fields.
+// mutateHook adds "omitempty" tag option to optional field from input type model as defined in graphql schema
 // For more info see https://github.com/99designs/gqlgen/blob/master/docs/content/recipes/modelgen-hook.md
-func mutateHook(b *modelgen.ModelBuild) *modelgen.ModelBuild {
-	for _, model := range b.Models {
-		for _, field := range model.Fields {
-			field.Tag = `json:"` + field.Name
-			if codegenconfig.IsNilable(field.Type) {
-				field.Tag += ",omitempty"
+func mutateHook(cfg *config.Config) func(b *modelgen.ModelBuild) *modelgen.ModelBuild {
+	return func(b *modelgen.ModelBuild) *modelgen.ModelBuild {
+		for _, model := range b.Models {
+			// only handle input type model
+			if schemaModel, ok := cfg.GQLConfig.Schema.Types[model.Name]; ok && schemaModel.IsInputType() {
+				for _, field := range model.Fields {
+					// find field in graphql schema
+					for _, def := range schemaModel.Fields {
+						if def.Name == field.Name {
+							// only add 'omitempty' on optional field as defined in graphql schema
+							if !def.Type.NonNull && codegenconfig.IsNilable(field.Type) {
```

Comment:
I'm not sure we need to keep `codegenconfig.IsNilable(field.Type)`

Reply:
oh! it is not needed!